services/headless-lms/models/src/exercise_slide_submissions.rs

```diff
@@ -191,10 +191,11 @@ WHERE id = $1
 /// Attempts to find a single random `ExerciseSlideSubmission` that is not related to the provided user.
 ///
 /// This function is mostly provided for very specific peer review purposes.
-pub async fn try_to_get_random_from_other_users_by_exercise_id(
+pub async fn try_to_get_random_filtered_by_user_and_submissions(
     conn: &mut PgConnection,
     exercise_id: Uuid,
     excluded_user_id: Uuid,
+    excluded_ids: &[Uuid],
 ) -> ModelResult<Option<ExerciseSlideSubmission>> {
     let res = sqlx::query_as!(
         ExerciseSlideSubmission,
```

Comment:
Add TODO here because we don't want to add people to the queue before they have given enogh reviews.